apps/emqx/src/emqx_authentication.erl

```diff
@@ -215,17 +215,20 @@ when
 %%------------------------------------------------------------------------------
 
 authenticate(#{enable_authn := false}, _AuthResult) ->
+    inc_authenticate_metric('authentication.success.anonymous'),
     ignore;
 authenticate(#{listener := Listener, protocol := Protocol} = Credential, _AuthResult) ->
     case get_authenticators(Listener, global_chain(Protocol)) of
         {ok, ChainName, Authenticators} ->
             case get_enabled(Authenticators) of
                 [] ->
+                    inc_authenticate_metric('authentication.success.anonymous'),
                     ignore;
                 NAuthenticators ->
                     do_authenticate(ChainName, NAuthenticators, Credential)
```

Comment:
```
case do_authenticate(...) of
  a -> do_inc;
  b -> do_inc
end
```

Reply:
Yes. it is better increasing metrics here